NBitcoin/RPC/RPCClient.cs

```diff
@@ -987,18 +987,18 @@ public Block GetBlock(int height)
 		public BlockHeader GetBlockHeader(uint256 blockHash)
 		{
 			var resp = SendCommand("getblockheader", blockHash.ToString());
-			return ParseBlockHeader(resp);
+			return ParseBlockHeader(resp, Network.Consensus.BlockHeaderFactory.CreateNewBlockHeader());
 		}
 
 		public async Task<BlockHeader> GetBlockHeaderAsync(uint256 blockHash)
 		{
 			var resp = await SendCommandAsync("getblockheader", blockHash.ToString()).ConfigureAwait(false);
-			return ParseBlockHeader(resp);
+			return ParseBlockHeader(resp, Network.Consensus.BlockHeaderFactory.CreateNewBlockHeader());
 		}
 
-		private static BlockHeader ParseBlockHeader(RPCResponse resp)
+		private static BlockHeader ParseBlockHeader(RPCResponse resp, BlockHeader blockheader)

```

Comment:
I think it would be easier to just remove `static` and directly use the factory inside.